components/entitlement/org.wso2.carbon.identity.entitlement/src/main/java/org/wso2/carbon/identity/entitlement/EntitlementPolicyAdminService.java

```diff
@@ -764,12 +763,6 @@ private void addOrUpdatePolicy(PolicyDTO policyDTO, boolean isAdd) throws Entitl
                 } else {
                     throw new EntitlementException("Unsupported Entitlement Policy. Policy can not be parsed");
                 }
-                try {
-                    String version = versionManager.createVersion(policyDTO);
```

Comment:
Policy versioning is removed from this layer and moved to DAO level impl (`PolicyDAO.addOrUpdatePolicy()`) itself. Validate whether this is ok.

Reply:
`PAPPolicyStoreManager.addOrUpdatePolicy()` getting called from three other places.

`EntitlementUtil.addFilesystemPolicy()` [behaves similarly like here](https://github.com/wso2/carbon-identity-framework/pull/5686/files#diff-fe29d56435d19e89e91e764fcaa98fcb0a60bff15df626c68c52206d4892de5cL455).
ie: Creating new version removed from the method impl and new version creation now handled in PolicyDAO.addOrUpdatePolicy().

But two other places (mentioned below) does not creates a version as of now. But with the proposed change it will creates a new version.
- `EntitlementPolicyAdminService.orderPolicy()`
- `EntitlementPolicyAdminService.enableDisablePolicy()`

Reply:
This causes a behavioral change where a new policy version is created when updating the policy order or enabling/disabling a published policy from PDP. To address, we can make versioning a conditional process.

Reply:
Made versioning in `addOrUpdatePolicy` conditional.